Touch interaction: allow users to disable touch support completely

josephsl opened the pull request:
### Link to issue number:
Fixes #9682 

### Summary of the issue:
Allow users to turn off touch support completely.

### Description of how this pull request fixes the issue:
Added a checkbox (and a corresponding flag in NVDA configuration database) that disables touch support completely. If disabled, touchscreen will behave as though NVDA is not running. Also, added support for touch being disabled on per-profile basis i.e. touch can be disabled if an app comes with its own touch interaction model.

### Testing performed:
Tested via try builds on a laptop with touch capabilities.

### Known issues with pull request:
None so far.

### Change log entry:
Changes (although this can go as a new feature):

You can now enable or disable NVDA's touchscreen support. This is done by checking a new checkbox in Touch Interaction panel of NVDA settings. (#9682)

Thanks.

josephsl committed:
ConfSpec: add touch enabled key. Re #9682.

Foundation: allow users to disable touch handler overlay by adding 'touch enabled' key to NVDA settings.

josephsl committed:
Touch handler: copyright header

josephsl committed:
Touch handler: use platform version tuple to determine if touch support should be used. Re #9682

josephsl committed:
Touch handler: enable touch support/overlay if told to do so by users. re #9682.

Enable touch support if configured to do so via settings. Also, change log text on touchscreen detection to just announce how many touch points are available before proceeding to enable/disable touch support.

josephsl committed:
Settings/touch interaction: add touch support toggle. Re #9682.

Allow users to enable/disable touch support via Touch Interaction settings panel.

josephsl committed:
Touch supporte3d: add debugLog flag so excessive log output will not be seen unless requested. Re #9682.

josephsl committed:
Touch handler: handle configuration profile switches via a dedicated touch mode setter function. Re #9682.

Similar to audio ducking support: use a dedicated 'set touch support' function to enable/disable touch handler thread based on whether touch support should be anbeld or not. This function in turn will be called when configuration profile changes to support scenarios such as manually turning off touch support or using profiles for apps where NVDA's own touch support should be turned off.

josephsl committed:
Touch handler: set touch support mode and register/unregister profile switching support. Re #9682.

Touch handler's initialize function will set appropriate touch support mode based on current config value. Also, both initialize and termiante methods support profile switching by calling register/unregister function, respectively.

josephsl committed:
Touch interaction settings panel: use touchHandler.setTouchSupport function to toggle touch handler thread. Re #9682.

josephsl committed:
Touch interaction panel: add accelerator.

josephsl committed:
User guide: add notes on touch interaction support checkbox in settings. Re #9682

josephsl commented:
Hi,

This will be considered "draft" while NVDA 2019.3 is under development; once stable version is released, draft status will be removed. IN the meantime, comments (including design comments) are appreciated.

Thanks.

leonardder commented on source/touchHandler.py:
```diff
@@ -294,30 +293,56 @@ def notifyInteraction(self, obj):
 
-def touchSupported():
+
+def touchSupported(debugLog=False):
```
```suggestion
def touchSupported(debugLog: bool = False):
```

leonardder commented on source/touchHandler.py:
```diff
@@ -294,30 +293,56 @@ def notifyInteraction(self, obj):
-def touchSupported():
+
+def touchSupported(debugLog=False):
 	"""Returns if the system and current NVDA session supports touchscreen interaction.
```
```suggestion
	"""Returns if the system and current NVDA session supports touchscreen interaction.
	@param debugLog: Whether to log additional details about touch support to the NVDA log.
```

leonardder commented on source/touchHandler.py:
```diff
@@ -294,30 +293,56 @@ def notifyInteraction(self, obj):
+		log.debug("Touch support disabled.")
+
+
+def handlePostConfigProfileSwitch():
```
Do we execute a config profile switch when saving config from the gui? I recall we don't.

leonardder commented on source/touchHandler.py:
```diff
@@ -294,30 +293,56 @@ def notifyInteraction(self, obj):
 
-def initialize():
+
+def setTouchSupport(enable):
```
```suggestion
def setTouchSupport(enable: bool):
```

LeonarddeR reviewed (changes requested):
Could you please change the touch panel to guiHelper while at it?

josephsl commented:
Hi, I don’t think we execute profile switch handlers when settings are saved – I borrowed this logic from Speech category/audio ducking. Thanks.

 

From: Leonard de Ruijter <notifications@github.com> 
Sent: Thursday, November 28, 2019 7:50 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Assign <assign@noreply.github.com>
Subject: Re: [nvaccess/nvda] Touch interaction: allow users to disable touch support completely (#10557)

 

@leonardder requested changes on this pull request.

Could you please change the touch panel to guiHelper while at it?

  _____  

In source/touchHandler.py <https://github.com/nvaccess/nvda/pull/10557#discussion_r351847589> :

> @@ -294,30 +293,56 @@ def notifyInteraction(self, obj):
 
 
 
 handler=None
 
 
 
-def touchSupported():
 
+
 
+def touchSupported(debugLog=False):
 

⬇️ Suggested change 

-def touchSupported(debugLog=False):
 
+def touchSupported(debugLog: bool = False):
 
  _____  


In source/touchHandler.py <https://github.com/nvaccess/nvda/pull/10557#discussion_r351847860> :

> @@ -294,30 +293,56 @@ def notifyInteraction(self, obj):
 
 
 
 handler=None
 
 
 
-def touchSupported():
 
+
 
+def touchSupported(debugLog=False):
 
        """Returns if the system and current NVDA session supports touchscreen interaction.
 

⬇️ Suggested change 

-       """Returns if the system and current NVDA session supports touchscreen interaction.
 
+       """Returns if the system and current NVDA session supports touchscreen interaction.
 
+       @param debugLog: Whether to log additional details about touch support to the NVDA log.
 
  _____  


In source/touchHandler.py <https://github.com/nvaccess/nvda/pull/10557#discussion_r351848365> :

>       global handler
 
        if not touchSupported():
 
               raise NotImplementedError
 
-       handler=TouchHandler()
 
-       log.debug("Touch support initialized. maximum touch inputs: %d"%windll.user32.GetSystemMetrics(SM_MAXIMUMTOUCHES))
 
+       if not handler and enable:
 
+              handler = TouchHandler()
 
+              log.debug("Touch support enabled.")
 
+       elif handler and not enable:
 
+              handler.terminate()
 
+              handler = None
 
+              log.debug("Touch support disabled.")
 
+
 
+
 
+def handlePostConfigProfileSwitch():
 

Do we execute a config profile switch when saving config from the gui? I recall we don't.

  _____  

In source/touchHandler.py <https://github.com/nvaccess/nvda/pull/10557#discussion_r351848669> :

>              return False
 
        return True
 
 
 
-def initialize():
 
+
 
+def setTouchSupport(enable):
 

⬇️ Suggested change 

-def setTouchSupport(enable):
 
+def setTouchSupport(enable: bool):
 

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/10557?email_source=notifications&email_token=AB4AXEENNEBHTSM7KEEUUDLQV7SBDA5CNFSM4JSWFFKKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCNKW35Q#pullrequestreview-324365814> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB4AXED5YJMTMZFKERBPEBTQV7SBDANCNFSM4JSWFFKA> .

josephsl committed:
Touch handler: review comments (annotations, docstring). Re #9682.

Reviewed by Leonard de Ruijter: add type annotations, along with docstring for touch supported function.

josephsl committed:
Touch interaction panel: convert to use GUI helper. re #9682.

Comment from Leonard de Ruijter: convert touch interaction panel to use GUI helper code just like other panels.

leonardder commented on user_docs/en/userGuide.t2t:
```diff
@@ -1494,6 +1495,11 @@ If you check this option and you have the "Enable mouse tracking" option enabled
+==== Enable touch interaction support ====[TouchSupportEnable]
+This checkbox enables NVDA's touch interaction support.
+If enabled, you can use your fingers to navigate and interact with items on screen using a touchscreen device.
+If disabled, touchscreen support from NVDA will be disabled.
```
What does this mean?

```suggestion
If disabled, touchscreen support from NVDA will be disabled, making the touch screen behave as normal.
```

josephsl commented:
Hi, I had difficulty when writing that sentence, and the suggestion above confirms my thought that we need to clarify a bit. Thanks for pointing this out.

 

From: Leonard de Ruijter <notifications@github.com> 
Sent: Thursday, November 28, 2019 10:09 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Assign <assign@noreply.github.com>
Subject: Re: [nvaccess/nvda] Touch interaction: allow users to disable touch support completely (#10557)

 

@leonardder commented on this pull request.

  _____  

In user_docs/en/userGuide.t2t <https://github.com/nvaccess/nvda/pull/10557#discussion_r351991498> :

> @@ -1494,6 +1495,11 @@ If you check this option and you have the "Enable mouse tracking" option enabled
 
 This settings category, only available on computers running Windows 8 and later with touch capabilities, allows you to configure how NVDA interacts with touchscreens.
 
 This category contains the following options:
 
 
 
+==== Enable touch interaction support ====[TouchSupportEnable]
 
+This checkbox enables NVDA's touch interaction support.
 
+If enabled, you can use your fingers to navigate and interact with items on screen using a touchscreen device.
 
+If disabled, touchscreen support from NVDA will be disabled.
 

What does this mean?

⬇️ Suggested change 

-If disabled, touchscreen support from NVDA will be disabled.
 
+If disabled, touchscreen support from NVDA will be disabled, making the touch screen behave as normal.
 

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/10557?email_source=notifications&email_token=AB4AXEFQ4HCRZ2FCAXCLS3TQWCWYDA5CNFSM4JSWFFKKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCNMAHRQ#pullrequestreview-324535238> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB4AXEDQD3FWECCLQQC7LQ3QWCWYDANCNFSM4JSWFFKA> .

josephsl committed:
User guide: clarify the behavior of touch support when touch is disabled. Re #9682

josephsl committed:
Touch handler: update copyright years

josephsl commented:
Hi,

Now that 2019.3 is under translatable string freeze, and with 2020.1 changelog showing, I'd like to get this PR reviewed again.

Thanks.

josephsl commented:
Hi,

Suspended until further notice in light of #11006. 

Thanks.

feerrenrut commented:
> Suspended until further notice in light of #11006.

Please continue to work on this PR. #11006 requests no new PR's. This PR already exists and is thus in the backlog of PR's that we want to address before accepting new ones. We are trying to stop PR's from going stale, and then becoming hard to merge. 

Forgive me for replying to you multiple times, I want to avoid the community being mislead.

josephsl commented:
Hi, I’ll take a look at this PR as early as Thursday. Thanks.

 

From: Reef Turner <notifications@github.com> 
Sent: Wednesday, April 15, 2020 3:52 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Assign <assign@noreply.github.com>
Subject: Re: [nvaccess/nvda] Touch interaction: allow users to disable touch support completely (#10557)

 

Suspended until further notice in light of #11006 <https://github.com/nvaccess/nvda/issues/11006> .

Please continue to work on this PR. #11006 <https://github.com/nvaccess/nvda/issues/11006>  requests no new PR's. This PR already exists and is thus in the backlog of PR's that we want to address before accepting new ones. We are trying to stop PR's from going stale, and then becoming hard to merge.

Forgive me for replying to you multiple times, I want to avoid the community being mislead.

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/10557#issuecomment-613967317> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB4AXEBSOCC6PINIURLBJJTRMWGV5ANCNFSM4JSWFFKA> .

dingpengyu commented:
> Hi,
> 
> Suspended until further notice in light of #11006.
> 
> Thanks.

hello josephsl
You currently have the following pr not merged yet:
#10990 
#10922 
#10378 
#10349 
#9429 
#9119 
#8644 
#8457 
and
#6930 
Nine in total pull request.
In addition to the current COVID-19 epidemic, there is ample time.

josephsl commented:
Hi, I’ll evaluate each of those and comment on status of these soon – rest assured that I’m thinking about them. Thanks for the reminder.

 

From: eric <notifications@github.com> 
Sent: Thursday, April 23, 2020 3:00 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Assign <assign@noreply.github.com>
Subject: Re: [nvaccess/nvda] Touch interaction: allow users to disable touch support completely (#10557)

 

Hi,

Suspended until further notice in light of #11006 <https://github.com/nvaccess/nvda/issues/11006> .

Thanks.

hello josephsl
You currently have the following pr not merged yet:
#10990 <https://github.com/nvaccess/nvda/pull/10990> 
#10922 <https://github.com/nvaccess/nvda/pull/10922> 
#10378 <https://github.com/nvaccess/nvda/pull/10378> 
#10349 <https://github.com/nvaccess/nvda/pull/10349> 
#9429 <https://github.com/nvaccess/nvda/pull/9429> 
#9119 <https://github.com/nvaccess/nvda/pull/9119> 
#8644 <https://github.com/nvaccess/nvda/pull/8644> 
#8457 <https://github.com/nvaccess/nvda/pull/8457> 
and
#6930 <https://github.com/nvaccess/nvda/pull/6930> 
Nine in total pull request.
In addition to the current COVID-19 epidemic, there is ample time.

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/10557#issuecomment-618306897> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB4AXEETWXRRBDKDPBPERM3ROAGRHANCNFSM4JSWFFKA> .

JulienCochuyt commented:
@josephsl, thanks for taking care of this. The lack of this feature has been annoying me quite a lot lately.

My few grains of salt on this topic...
- Would you please consider allowing to bind a gesture on toggling touch screen support on and off?
- Shouldn't we consider to temporarily disable touch screen support when pressing `NVDA+F2`? If not, any suitable mean of letting a touch gesture through would be really handy.
- Does UIA or IA2 expose the fact that a given control should have touch gestures not intercepted by AT? (such as found on touch musical instruments in IOS) If so, we should most likely plan on honoring it.
- Should we consider shipping a default profile for Windows Touch Magnifier window to disable touch support when interacting with it?

josephsl commented:
Hi,

Suspending touch support for passthrough command: not easy. In order to do so, NVDA must be able to detect that touch interaction has occurred after it is suspended. This isn't quite possible because once an assistive technology registers for touch interaction, it won't let go unless it voluntarily gives up, and manual work is involved to bring touch support back online. Also, if multiple assistive technologies wants to work with touch interaction, only one will get a chance to do it (when it starts, Narrator presents a dialog about this when NVDA is running on a touchscreen computer).

Command to toggle touch support: although it makes sense, it isn't consistent. One can assign a keyboard or a braille command to toggle touch interaction. One can turn off touch interaction using touch gestures, but you can't bring it back with touch gestures as NVDA doesn't know if a gesture has occurred (same problem as above).

UIA/IA2 and touch support: probably not unless I read the documentation wrong.

As for Windows Magnifier window: one way to enhance this in the future is letting app modules or overlay classes define if touch interaction should be turned off. This is already possible with this PR through app-specific profiles, which users must specify manually.

Thanks.

JulienCochuyt commented:
@josephsl wrote:
> Suspending touch support for passthrough command: not easy. In order to do so, NVDA must be able to detect that touch interaction has occurred after it is suspended. This isn't quite possible because once an assistive technology registers for touch interaction, it won't let go unless it voluntarily gives up, and manual work is involved to bring touch support back online. 

Letting other ATs aside, do you mean there is no such mechanism as sending a touch gesture we received back to the control that would have received it if NVDA did not register?
If so, I think my behavior of choice would be to (at least optionally) suspend touch handling until the next keystroke, to allow for keyboard+touch usages.

> Command to toggle touch support: although it makes sense, it isn't consistent. One can assign a keyboard or a braille command to toggle touch interaction. One can turn off touch interaction using touch gestures, but you can't bring it back with touch gestures as NVDA doesn't know if a gesture has occurred (same problem as above).

Idem, I feel that inconsistency here is better than inability, especially if we cannot achieve a "next touch gesture pass through" command.

> As for Windows Magnifier window: one way to enhance this in the future is letting app modules or overlay classes define if touch interaction should be turned off. This is already possible with this PR through app-specific profiles, which users must specify manually.

Glad to read this. We for sure can take the time once this PR is merged to experiment with such a profile before deciding if it's worth being promoted as a default.

josephsl commented:
Hi,

Once an AT unregisters itself from getting touch gestures, it isn't possible to listen to touch gestures. This is because an invisible iwndow is used by the AT to listen to touch interaction signals, and suspending touch means destroying that window.

As for toggle command: I imagine a scenario where people would prefer to use touchscreen alone rather than using a keyboard along with the device (Surface Pro without type cover, for example).

Thanks.

JulienCochuyt commented:
> Once an AT unregisters itself from getting touch gestures, it isn't possible to listen to touch gestures. This is because an invisible iwndow is used by the AT to listen to touch interaction signals, and suspending touch means destroying that window.

Sure, but again, do you mean we cannot send back what we don't want to keep (as keyboard handler does)?

> As for toggle command: I imagine a scenario where people would prefer to use touchscreen alone rather than using a keyboard along with the device (Surface Pro without type cover, for example).

While this scenario makes sense, it's no reason IMHO to disallow altogether binding a keyboard gesture to do the toggling.

josephsl commented:
Hi, no, we can’t do that with touchscreen gestures at this time, as the window itself isn’t going to be used. In short, Windows will forget that an AT is active. As for toggle command bindings, it might be possible to do that at least for keyboards – it is something we should think about for a future iteration of this PR. Thanks.

 

From: Julien Cochuyt <notifications@github.com> 
Sent: Saturday, May 2, 2020 12:16 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Mention <mention@noreply.github.com>
Subject: Re: [nvaccess/nvda] Touch interaction: allow users to disable touch support completely (#10557)

 

Once an AT unregisters itself from getting touch gestures, it isn't possible to listen to touch gestures. This is because an invisible iwndow is used by the AT to listen to touch interaction signals, and suspending touch means destroying that window.

Sure, but again, do you mean we cannot send back what we don't want to keep (as keyboard handler does)?

As for toggle command: I imagine a scenario where people would prefer to use touchscreen alone rather than using a keyboard along with the device (Surface Pro without type cover, for example).

While this scenario makes sense, it's no reason IMHO to disallow altogether binding a keyboard gesture to do the toggling.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/10557#issuecomment-623000888> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB4AXEGQRUZRYDMDC7SBXW3RPRWP5ANCNFSM4JSWFFKA> .

josephsl committed:
Merge branch 'master' of https://github.com/nvaccess/nvda into i9682configurableTouchOverlay

michaelDCurran commented:
@JulienCochuyt  You bring up all good points, but these should be all handled in one or more separate PRs. Not only are they a little complex and will be somewhat technically unrelated to how this is currently implemented. By itself this current PR still has a clear use case. There are certainly low vision users who use NVDA to read text along with a magnifier, but have enough vision to interact with their touch screen independently and do not want NVDA ever intercepting touch. So @josephsl   assuming this passes review, I want to take this as soon as we can.

michaelDCurran commented:
@josephsl  just confirming that you have tested config profile switches where one profile has touch enabled and one does not? I.e. touch successfully gets re-enabled after being disabled by a previous profile?

josephsl commented:
Hi, glad to contribute another feature from Enhanced Touch Gestures to NVDA Core – thanks.

 

From: Michael Curran <notifications@github.com> 
Sent: Monday, June 22, 2020 6:14 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Mention <mention@noreply.github.com>
Subject: Re: [nvaccess/nvda] Touch interaction: allow users to disable touch support completely (#10557)

 

@JulienCochuyt <https://github.com/JulienCochuyt>  You bring up all good points, but these should be all handled in one or more separate PRs. Not only are they a little complex and will be somewhat technically unrelated to how this is currently implemented. By itself this current PR still has a clear use case. There are certainly low vision users who use NVDA to read text along with a magnifier, but have enough vision to interact with their touch screen independently and do not want NVDA ever intercepting touch. So @josephsl <https://github.com/josephsl>  assuming this passes review, I want to take this as soon as we can.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/10557#issuecomment-647848255> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB4AXEGVNRD32UCKBR252LDRX76V5ANCNFSM4JSWFFKA> .

josephsl commented:
Hi, yes, tested via Enhanced Touch Gestures add-on, which has identical construction. Thanks.

 

From: Michael Curran <notifications@github.com> 
Sent: Monday, June 22, 2020 6:24 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Mention <mention@noreply.github.com>
Subject: Re: [nvaccess/nvda] Touch interaction: allow users to disable touch support completely (#10557)

 

@josephsl <https://github.com/josephsl>  just confirming that you have tested config profile switches where one profile has touch enabled and one does not? I.e. touch successfully gets re-enabled after being disabled by a previous profile?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/10557#issuecomment-647851125> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB4AXEF2Q627FBO7S4GMNBTRX774VANCNFSM4JSWFFKA> .

josephsl commented:
Hi, forgot to say that profile switching does work. Thanks.

 

From: Michael Curran <notifications@github.com> 
Sent: Monday, June 22, 2020 6:24 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Mention <mention@noreply.github.com>
Subject: Re: [nvaccess/nvda] Touch interaction: allow users to disable touch support completely (#10557)

 

@josephsl <https://github.com/josephsl>  just confirming that you have tested config profile switches where one profile has touch enabled and one does not? I.e. touch successfully gets re-enabled after being disabled by a previous profile?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/10557#issuecomment-647851125> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB4AXEF2Q627FBO7S4GMNBTRX774VANCNFSM4JSWFFKA> .

michaelDCurran commented:
@leonardder  Are there any pending changes you are waiting for on this pr?

leonardder commented on source/touchHandler.py:
```diff
@@ -294,30 +293,57 @@ def notifyInteraction(self, obj):
 		return False
-	if (winVersion.winVersion.major*10+winVersion.winVersion.minor)<62:
-		log.debugWarning("Touch only supported on Windows 8 and higher")
+	if winVersion.winVersion.platform_version < (6, 2, 9200):
```
Is it really necessary to take the platform into account here?

```suggestion
	if (winVersion.winVersion.major, winVersion.winVersion.minor) < (6, 2):
```

LeonarddeR reviewed (approved):
If a sighted person can confirm that the gui looks good (though I don't have reason to think it wouldn't), I'm ok with this. Just one small thing.

josephsl commented:
Hi, the platform attribute gives us all the info we need in one function call, and it is also used for app module version lookup. Thanks.

 

From: Leonard de Ruijter <notifications@github.com> 
Sent: Tuesday, June 23, 2020 12:02 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Mention <mention@noreply.github.com>
Subject: Re: [nvaccess/nvda] Touch interaction: allow users to disable touch support completely (#10557)

 

@leonardder approved this pull request.

If a sighted person can confirm that the gui looks good (though I don't have reason to think it wouldn't), I'm ok with this. Just one small thing.

  _____  

In source/touchHandler.py <https://github.com/nvaccess/nvda/pull/10557#discussion_r444442481> :

>              return False
 
-       if (winVersion.winVersion.major*10+winVersion.winVersion.minor)<62:
 
-              log.debugWarning("Touch only supported on Windows 8 and higher")
 
+       if winVersion.winVersion.platform_version < (6, 2, 9200):
 

Is it really necessary to take the platform into account here?

⬇️ Suggested change 

-       if winVersion.winVersion.platform_version < (6, 2, 9200):
 
+       if (winVersion.winVersion.major, winVersion.winVersion.minor) < (6, 2):
 

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/10557#pullrequestreview-436070979> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB4AXEE4RBYES4TCND3DRMTRYD335ANCNFSM4JSWFFKA> .
